jenkins-job-components/src/main/java/com/ofg/job/component/JobComponent.java

```diff
@@ -0,0 +1,20 @@
+package com.ofg.job.component;
+
+import javaposse.jobdsl.dsl.Job;
+
+/**
+ * Interface for configurations applicable on Job.
+ */
+public interface JobComponent<J extends Job> {
+
+    JobComponent<? extends Job> EMPTY = job -> {};
+    
+    default <NJ extends J> JobComponent<NJ> and(JobComponent<NJ> other) {
```

Comment:
I managed to make it work by adding this to JobComponent:
```java
    default JobComponent<J> or(JobComponent<? super J> other) {
        return (job) -> {
            this.applyOn(job);
            other.applyOn(job);
        };
    }
```
But you have to believe me (and not the tests -unless you write them) for now :)
I wanted to use `>>` and `>>>` for the operators to overload, but so far the only operator that I found could be used this way was `or` and all the others raised compilation errors due to problems with type inference (probably a bug in groovy compiler).